### PR TITLE
Invite embeds

### PR DESCRIPTION
**Pull request by TaiAurori**
(This is my first contribution to a big code-base like this, so I request anyone reviewing be patient with me.)
This PR adds the ability for messages to show rich information about any invites that are contained inside them. 
This implementation is client-side and does not interact with January.
image

### CODE REVIEW

**Comment by Mysterious-Dev** [comment]
Maybe translate texts like _Joined_ and others with a translate key (for multilanguage).

File: src/components/common/messaging/Message.tsx [hunk 1]
```diff
@@ -142,6 +151,28 @@ const Message = observer(
...
+                                    inviteRegex.lastIndex = 0;
+                                    while ((match = inviteRegex.exec(content)) !== null) {
+                                        if (!results.includes(match[match.length - 1])) {
+                                            results.push(match[match.length - 1]);
+                                        }
+                                    }
+                                    return results.map(code => <EmbedInvite code={code} />);
+                                }
+                            }
+                        })()}
```
Comment (insertish) [inline src/components/common/messaging/Message.tsx:175]: I would throw this into its own component.

**Review by insertish — APPROVED** [review]
I will add an extra component for this myself during merge, ~~also add a limit to just the one invite per message~~.

**Comment by insertish** [comment]
Changed logic in 2ccc0b7b5e2de0ca54bae7571212f3d3516aff40.

**Pull request merged**